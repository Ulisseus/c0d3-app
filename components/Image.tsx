import * as React from 'react'
import Modal from 'react-bootstrap/Modal'
type Props = {
  src: string
}
const Image: React.FC<Props> = (props: any) => {
  const [show, setShow] = React.useState(false)
  return (
    <>
      <img className="MDX_image" {...props} onClick={() => setShow(true)} />
      <Modal classname="MDX_modal_image" show={show} centered>
        <Modal.Dialog>
          <Modal.Body>
            <img {...props} onClick={() => setShow(false)} />
          </Modal.Body>
        </Modal.Dialog>
      </Modal>
    </>
  )
}

export default Image
