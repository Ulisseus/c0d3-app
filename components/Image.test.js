import * as React from 'react'
import { render, screen } from '@testing-library/react'
import Image from './Image'
import userEvent from '@testing-library/user-event'

describe('Image component test', () => {
  test('Should render image', () => {
    const container = render(
      <Image src="/assets/curriculum/js-0-cover.svg" alt="test" />
    )
    expect(container).toMatchSnapshot()
    userEvent.click(screen.getByAltText('test'))
    expect(container).toMatchSnapshot()
    userEvent.click(screen.getAllByAltText('test')[1])
  })
})
