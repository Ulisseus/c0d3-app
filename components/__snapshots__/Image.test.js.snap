// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component test Should render image 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="test"
        class="MDX_image"
        src="/assets/curriculum/js-0-cover.svg"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="test"
      class="MDX_image"
      src="/assets/curriculum/js-0-cover.svg"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Image component test Should render image 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    style="padding-right: 0px;"
  >
    <div
      aria-hidden="true"
    >
      <img
        alt="test"
        class="MDX_image"
        src="/assets/curriculum/js-0-cover.svg"
      />
    </div>
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-dialog-centered"
        classname="MDX_modal_image"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-dialog"
          >
            <div
              class="modal-content"
            >
              <div
                class="modal-body"
              >
                <img
                  alt="test"
                  src="/assets/curriculum/js-0-cover.svg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <img
      alt="test"
      class="MDX_image"
      src="/assets/curriculum/js-0-cover.svg"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
